### DISCUSSION TITLE
Solving the experiment

### DISCUSSION THREAD

**Discussion by Jadhav-Sourabh** [pybamm-team/PyBaMM#2150] [General]
is this possible to simulate the experiment in a closed-loop manner? it will change the c-rate as some other real-time parameters changes.

**Comment by brosaplanella** [comment]
If I understand correctly, what you could do is to define a function that advances the simulation a small time interval given a C-rate and some parameter values. Using input parameters will help (see [this notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/parameterization/change-input-current.ipynb) for an example).

**Reply by Jadhav-Sourabh** [reply]
I want to use the currently running simulation parameter to decide the next current rate, like a closed-loop system. is this possible?
take temperature as a parameter to decide the current rate. e.g. cell temperature starting from 5 Celsius 0.5C and after some time it will cross 10 Celsius then I want to change current rate to 1C or 0.25C. 
is this possible to do?

**Reply by brosaplanella** [reply]
I think the easiest way is to do as I suggested above. The `step` method of the solvers (see [docs](https://pybamm.readthedocs.io/en/latest/source/solvers/base_solver.html#pybamm.BaseSolver.step) for more info) might be helpful.

**Reply by Jadhav-Sourabh** [reply]
This is not possible because we cannot find the temperature level in between of simulation, also we don't know about time interval means at which time the specific temperature is crossing.

**Reply by Jadhav-Sourabh** [reply]
with the help of only temperature, I want to solve this simulation by taking feedback of temperature

**Reply by brosaplanella** [reply]
But in practice you always define a time interval between samples, so you can do the same with the code (I mean a very short time interval, like one second). Alternatively, if you only change current based on a few thresholds, you can add try adding an event to the model to check when a temperature threshold is crossed and then change the C-rate and run the simulation again.
